torch/lib/THC/generic/THCTensorMath.cu

```diff
@@ -376,6 +376,28 @@ void THCTensor_(diag)(THCState *state, THCTensor *self_, THCTensor *src_, long k
   THCudaCheck(cudaGetLastError());
 }
 
+void THCTensor_(eye)(THCState *state, THCTensor *self_, long n, long m)
+{
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, self_));
+  THArgCheck(n > 0, 1, "invalid argument");
+
+  if(m <= 0)
+    m = n;
+
+  THCTensor_(resize2d)(state, self_, n, m);
+  THCTensor_(zero)(state, self_);
+
+  long sz = THMin(n, m);
+  long stride = THCTensor_(stride)(state, self_, 0) +
```

Comment:
Not necessarily. This implementation handles cases where the output tensor is non-contiguous. For example
```python
a = torch.cuda.FloatTensor(3, 4).t()
torch.eye(4, 4, out=a)
```